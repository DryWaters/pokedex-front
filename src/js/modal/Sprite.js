import React from 'react';
import PropTypes from 'prop-types';

import { getPokemonSprite } from '../core/PokemonAPI';

import '../../styles/modal/Sprite.css';
import '../../styles/core/PokemonType.css';

const propTypes = {
  name: PropTypes.string.isRequired,
  image_path: PropTypes.string.isRequired,
  types: PropTypes.arrayOf(PropTypes.string).isRequired,
};

class Sprite extends React.Component {
  render () {
    const inner = (
      <img className="modal-sprite" src={getPokemonSprite(this.props.image_path)} alt={`${this.props.name} sprite`} />
    );

    return (
      <div className={this.props.className}>
        <div className={`sprite-border primary ${this.props.types[0]}-border`}>
          {this.props.types[1] ?
            <div className={`sprite-border secondary ${this.props.types[1]}-border`}>
              {inner}
            </div>
          : <div className="single-type-img" > {inner} </div>}
        </div>
        <h2> {this.props.name} </h2>
      </div>
    );
  }

}

Sprite.propTypes = propTypes;

export default Sprite;
