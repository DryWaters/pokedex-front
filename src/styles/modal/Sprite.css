.sprite-border {
  border-style: solid;
  border-width: 3px;
  border-radius: 20px;
}

.primary {
  width: 380px;
  height:380px;
}

.secondary {
  width: 360px;
  height: 360px;
  transform: translateY(7px);
}

.single-type-img {
  transform: translateY(12px);
}

.modal-sprite {
  max-height: 350px;
  max-width: 350px;
}
