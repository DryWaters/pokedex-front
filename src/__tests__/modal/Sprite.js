import React from 'react';
import { shallow } from 'enzyme';

import Sprite from '../../js/modal/Sprite';

import ExampleJSON from '../../../local_data/example_detail.json';

describe('Sprite Component', () => {
  let props;
  beforeEach(() => {
    props = {
      name: ExampleJSON.name,
      image_path: ExampleJSON.forms[0].image_path,
      types: ExampleJSON.forms[0].types,
    };
  });

  it('should render Sprite component without crashing', () => {
    shallow(<Sprite {...props} />);
  });

  it('should match Sprite component snapshot', () => {
    const wrapper = shallow(<Sprite {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

});
