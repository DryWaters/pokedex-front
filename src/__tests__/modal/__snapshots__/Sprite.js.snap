// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sprite Component should match Sprite component snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sprite
    image_path="/sprites/pokemon/large/12.png"
    name="Butterfree"
    types={
        Array [
            "bug",
            "flying",
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="sprite-border primary bug-border"
>
          <div
                    className="sprite-border secondary flying-border"
          >
                    <img
                              alt="Butterfree sprite"
                              className="modal-sprite"
                              src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png"
                    />
          </div>
</div>,
        <h2>
           
          Butterfree
           
</h2>,
      ],
      "className": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="sprite-border secondary flying-border"
>
            <img
                        alt="Butterfree sprite"
                        className="modal-sprite"
                        src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png"
            />
</div>,
          "className": "sprite-border primary bug-border",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Butterfree sprite"
              className="modal-sprite"
              src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png"
/>,
            "className": "sprite-border secondary flying-border",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Butterfree sprite",
              "className": "modal-sprite",
              "src": "https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " ",
            "Butterfree",
            " ",
          ],
        },
        "ref": null,
        "rendered": Array [
          " ",
          "Butterfree",
          " ",
        ],
        "type": "h2",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="sprite-border primary bug-border"
>
            <div
                        className="sprite-border secondary flying-border"
            >
                        <img
                                    alt="Butterfree sprite"
                                    className="modal-sprite"
                                    src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png"
                        />
            </div>
</div>,
          <h2>
             
            Butterfree
             
</h2>,
        ],
        "className": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="sprite-border secondary flying-border"
>
              <img
                            alt="Butterfree sprite"
                            className="modal-sprite"
                            src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png"
              />
</div>,
            "className": "sprite-border primary bug-border",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Butterfree sprite"
                className="modal-sprite"
                src="https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png"
/>,
              "className": "sprite-border secondary flying-border",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Butterfree sprite",
                "className": "modal-sprite",
                "src": "https://pokedex-backend-server.herokuapp.com/sprites/pokemon/large/12.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " ",
              "Butterfree",
              " ",
            ],
          },
          "ref": null,
          "rendered": Array [
            " ",
            "Butterfree",
            " ",
          ],
          "type": "h2",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
